app/models/concerns/csv_orders_parser.rb

```
@@ -0,0 +1,89 @@
+        end
+      end
+    rescue TransactionRollbackError
+      return {consignee_address: consignee_address, consignee_contact: consignee_contact,
+        consignor_address: consignor_address, consignor_contact: consignor_contact, order_release: order_release}
+    end
+
+    nil
+  end
+
+  private
+
+  def save_consignee(row)
+    address_row = remap_hash_keys(row,
+      {"destination_state" => :state, "destination_zip" => :zip, "destination_raw_line_1" => :raw_line,
+       "destination_city" => :city})
+    address = Address.find_or_create_by(address_row)
+
+    contact_row = remap_hash_keys(row, {"client name" => :name})
+    contact_row[:address] = address
+    contact = Contact.find_or_create_by(contact_row)
+
+    return address, contact
+  end
+
+  def save_consignor(row)
+    address_row = remap_hash_keys(row,
+      {"origin_state" => :state, "origin_zip" => :zip, "origin_raw_line_1" => :raw_line, "origin_city" => :city})
+    address = Address.find_or_create_by(address_row)
+
+    contact_row = remap_hash_keys(row, {"origin_name" => :name})
+    contact_row[:address] = address
+    contact = Contact.find_or_create_by(contact_row)
+
+    return address, contact
+  end
+
+  def save_order_release(row, warehouse, consignee, consignor)
+    order_release_row = remap_hash_keys(row,
+      {"purchase_order_number" => :purchase_order_number, "volume" => :volume, "delivery_shift" => :delivery_shift,
+       "handling_unit_quantity" => :handling_unit_quantity})
+    order_release_row[:responsible_warehouse] = warehouse
+    order_release_row[:consignee] = consignee
+    order_release_row[:consignor] = consignor
+    order_release_row[:delivery_date] = Date.strptime(row["delivery_date"], "%m/%d/%Y") rescue nil
+    handling_unit_type = HandlingUnitType.by_name(row["handling_unit_type"]).first
+    order_release_row[:handling_unit_type] = handling_unit_type unless handling_unit_type.nil?
+    OrderRelease.create(order_release_row)
```

As I understood, OrderRelease has a validation for delivery_date, so OrderRelease.create(order_release_row) will silently fail, is it ok?